src/codegen/compile.ml

```diff
@@ -3980,6 +4095,113 @@ module StableMem = struct
           get_n32 ^^
           write_word32 env)
     | _ -> assert false
+
+  let load_word8 env =
+    match E.mode env with
+    | Flags.ICMode | Flags.RefMode ->
+      Func.share_code1 env "__stablemem_load_word8"
+        (("offset", I32Type)) [I32Type]
+        (fun env get_offset  ->
+          get_offset ^^
+          guard env ^^
```

Comment:
`guard env` could leave the offset on the stack, then the next line is not necessary.

Reply:
That's just pushing the load elsewhere and I'd like to keep the code similar to guard_range...